lib/chef/cookbook/gem_installer.rb

```diff
@@ -45,10 +46,16 @@ def install
 
         unless cookbook_gems.empty?
           begin
-            inline_gemfile do
-              source Chef::Config[:rubygems_url]
-              cookbook_gems.each do |args|
-                gem(*args)
+            Dir.mktmpdir("chef-gem-bundle") do |dir|
+              File.open("#{dir}/Gemfile", "w+") do |tf|
+                tf.puts "source '#{Chef::Config[:rubygems_url]}'"
+                cookbook_gems.each do |args|
+                  tf.puts "gem #{args.map { |i| "'#{i}'" }.join(' ')}"
+                end
+                tf.close
+                Dir.chdir(dir) do
+                  so = shell_out!("bundle install")
```

Comment:
I think we probably do?  Maybe via the omnibus-detection stuff inside the rubygems provider that we use for chef_gem and then run bundler from that path...